code/build.gradle

```diff
@@ -1,53 +1,43 @@
 // Top-level build file where you can add configuration options common to all sub-projects/modules.
-buildscript {
-    ext {
-        kotlin_version = '1.5.0'
-    }
-    repositories {
-        google()
-        mavenCentral()
-        maven { url "https://plugins.gradle.org/m2/" }
-        maven { url "https://oss.sonatype.org/content/repositories/snapshots/" }
-    }
-    dependencies {
-        classpath 'com.android.tools.build:gradle:4.0.1'
-        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
-        classpath "com.diffplug.spotless:spotless-plugin-gradle:5.12.5"
-        // NOTE: Do not place your application dependencies here; they belong
-        // in the individual module build.gradle files
-    }
-}
-
-allprojects {
-    repositories {
-        google()
-        mavenCentral()
-        maven { url "https://plugins.gradle.org/m2/" }
-        maven { url "https://oss.sonatype.org/content/repositories/snapshots/" }
-    }
-
-    apply plugin: 'maven-publish'
-}
 
-task clean(type: Delete) {
-    delete rootProject.buildDir
+plugins {
+    id 'com.android.application' version '7.3.0' apply false
+    id 'com.android.library' version '7.3.0' apply false
+    id 'com.diffplug.gradle.spotless' version '6.11.0' apply false
+    id 'org.jetbrains.kotlin.android' version "1.6.21" apply false
 }
 
 ext {
     // android config
     minSdkVersion = 19
-    targetSdkVersion = 29
-    compileSdkVersion = 29
+    targetSdkVersion = 33
+    compileSdkVersion = 33
     versionCode = 1
 
     // java config
-    sourceCompatibility = JavaVersion.VERSION_1_7
-    targetCompatibility = JavaVersion.VERSION_1_7
+    sourceCompatibility = JavaVersion.VERSION_1_8
+    targetCompatibility = JavaVersion.VERSION_1_8
 
-    // dependencies
-    junitVersion = "1.1.2"
+    // kotlin config
+    kotlinVersion = "1.6.21"
+    kotlinJvmTarget = "1.8"
+
+    // test dependencies
+    junitVersion = "1.1.3"
     mockitoCoreVersion = "2.28.2"
-    buildToolsVersion = "29.0.3"
     testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"
-    jacocoVersion = "0.8.5"
+
+    // build tools
+    buildToolsVersion = "29.0.3"
```

Comment:
Setting the build tools version shouldn't be required anymore, can you remove this?